metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlSavePartition.java

```diff
@@ -135,6 +135,13 @@ private void _insert(final QualifiedName tableQName, final Table table, final Ta
             for (int i = 0; i < partValues.size(); i++) {
                 partitionKeyValsValues.add(new Object[] {partId, partValues.get(i), i});
             }
+            // Partition parameters
+            final Map<String, String> parameters = partition.getMetadata();
+            if (parameters != null) {
```

Comment:
more of an all encompassing suggestion but we should start using `Optional` to reflect nullable condition as an API/module delineation level. I doubt you'll want to change this now but I think it makes for more manageable code rather than guessing or having to always check documentation to see if something is null.